charms/knative-eventing/tests/unit/test_charm.py

```diff
@@ -1,67 +1,64 @@
+    mocked_apply = mocker.patch(
+        "k8s_resource_handler.kubernetes._kubernetes_resource_handler.KubernetesResourceHandler.apply"
+    )
+    mocked_apply.return_value = None
 
-        self.assertTrue(action_event.set_results.called)
+    harness.begin()
+    harness.charm.on.install.emit()
+    mocked_apply.assert_called()
+    assert harness.model.unit.status == ActiveStatus()
 
-    def test_action_fail(self):
-        action_event = Mock(params={"fail": "fail this"})
-        self.harness.charm._on_fortune_action(action_event)
 
-        self.assertEqual(action_event.fail.call_args, [("fail this",)])
+@pytest.mark.parametrize(
+    "apply_error, raised_exception, expected_blocked_message",
+    (
+        (FakeApiError(400), pytest.raises(FakeApiError), "ApiError: 400"),
+        (
+            FakeApiError(403),
+            pytest.raises(ReconcileError),
+            "ErrorWithStatus: broken",
+        ),
+    ),
+)
+def test_on_install_blocked(
+    apply_error,
+    raised_exception,
+    expected_blocked_message,
+    harness,
+    mocked_lightkube_client,
+    mocker,
+):
+    mocked_apply = mocker.patch(
+        "k8s_resource_handler.kubernetes._kubernetes_resource_handler.KubernetesResourceHandler.apply"
+    )
+    mocked_apply.side_effect = apply_error
 
-    def test_httpbin_pebble_ready(self):
-        # Check the initial Pebble plan is empty
-        initial_plan = self.harness.get_container_pebble_plan("httpbin")
-        self.assertEqual(initial_plan.to_yaml(), "{}\n")
-        # Expected plan after Pebble ready with default config
-        expected_plan = {
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": "🎁"},
-                }
-            },
-        }
-        # Get the httpbin container from the model
-        container = self.harness.model.unit.get_container("httpbin")
-        # Emit the PebbleReadyEvent carrying the httpbin container
-        self.harness.charm.on.httpbin_pebble_ready.emit(container)
-        # Get the plan now we've run PebbleReady
-        updated_plan = self.harness.get_container_pebble_plan("httpbin").to_dict()
-        # Check we've got the plan we expected
-        self.assertEqual(expected_plan, updated_plan)
-        # Check the service was started
-        service = self.harness.model.unit.get_container("httpbin").get_service("httpbin")
-        self.assertTrue(service.is_running())
-        # Ensure we set an ActiveStatus with no message
-        self.assertEqual(self.harness.model.unit.status, ActiveStatus())
+    harness.begin()
+    with raised_exception:
+        harness.charm.on.install.emit()
+    assert harness.model.unit.status == BlockedStatus(expected_blocked_message)
```

Comment:
Cory gave me this tip a while back and I sometimes agree, sometimes disagree.  He thought checking for a specific message in something like this was fragile (any small things will break it), whereas just asserting `assert isinstance(harness.model.unit.status, BlockedStatus)` was more robust.  Depends on what you want to accomplish, but you can consider it